src/main/java/org/opensearch/knn/plugin/stats/CircuitBreakerStat.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.stats;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.knn.plugin.transport.KNNCircuitBreakerTrippedAction;
+import org.opensearch.knn.plugin.transport.KNNCircuitBreakerTrippedRequest;
+import org.opensearch.transport.client.Client;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public class CircuitBreakerStat extends KNNStat<Boolean> {
```

Comment:
TODO: In mixed cluster state, this should fall back to reading from the (deprecated) index setting.

Reply:
I fixed this in latest revision